test/message_extraction/message_extraction_test.dart

```diff
@@ -103,22 +105,28 @@ void copyFilesToTempDirectory() {
     asTestDirPath('mock_flutter/services.dart'),
   ];
 
-  for (var filename in files) {
-    var file = new File(filename);
+  for (var filePath in files) {
+    var file = new File(filePath);
     if (file.existsSync()) {
-      file.copySync(path.join(tempDir, path.basename(filename)));
+      file.copySync(path.join(tempDir, path.basename(filePath)));
     }
   }
 
-  // Copy our package_config.json file so the test can locate packages.
-  var sourcePackageConfig =
-      File(path.join('.dart_tool', 'package_config.json'));
-  var destPackageConfig =
-      File(path.join(tempDir, '.dart_tool', 'package_config.json'));
-  if (!destPackageConfig.parent.existsSync()) {
-    destPackageConfig.parent.createSync();
+  // TODO:devoncarew): Improve the integration testing story here.
+  // Here we copy the package config file so the test can locate packages.
+  // For github, we use package_config.json; for google3, .packages.
+  final dartTool = '.dart_tool';
+
+  var configFile = File.fromUri(await Isolate.packageConfig);
+  var destFile = File(path.joinAll([
+    tempDir,
+    if (configFile.path.contains(dartTool)) dartTool,
+    path.basename(configFile.path),
+  ]));
```

Comment:
```suggestion
  var configFile = File.fromUri(await Isolate.packageConfig);
  var destFile = File(path.joinAll([
    tempDir,
    '.dartTool',
    'package_config.json',
  ]));
```

As far as I know we don't need to keep an identical layout. The file will contain json either way.

Reply:
That works (in both contexts)! Updated to always copy into package_config.json (w/ slightly different code that above).